### PR TITLE
feat: add checks for standard and proprietary header usage (#591)

### LINKED ISSUE #591: Reflect Http Headers-Related Changes in the Guidelines
There was an initiative to clean up HTTP header-related guidelines (https://github.com/zalando/restful-api-guidelines/issues/297). As result, the rule `[131] Use Hyphenated HTTP Headers` has been removed. We have to reflect the changes in Zally.

**Status Quo**: Guidelines state, that we **must** use either standard HTTP headers or the ones which are defined in the [[183]](http://zalando.github.io/restful-api-guidelines/#183).

I propose to:
- Delete `[131] Use Hyphenated HTTP Headers` rule
- Implemente a new whitelist-based rule for [[183]](http://zalando.github.io/restful-api-guidelines/#183) which checks: `∃header. header ∉ {standard-headers} ∪ {zalando-headers} `

### PR DESCRIPTION
**Pull request by maxim-tschumak**
Closes #591

### CODE REVIEW

**Comment by maxim-tschumak** [comment]
@ddufourboivin-zalando I've incorporated (almost all) your comments and rebased the branch - please have another look

File: server/src/main/java/de/zalando/zally/rule/zalando/ProprietaryHeadersRule.kt [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            .filterNot { it.name.toLowerCase() in requestHeaders }
+            .map { context.violation(requestDescription, it) }
+
+    @Check(severity = Severity.MUST)
+    fun validateResponseHeaders(context: Context): List<Violation> = responseHeaders(context)
+            .filterNot { it.key.toLowerCase() in responseHeaders }
+            .map { context.violation(responseDescription, it.value) }
+
+    private fun requestHeaders(context: Context): List<Parameter> = context.api.paths.values
+            .flatMap { it.readOperations().flatMap { it.parameters.orEmpty().filter { "header" == it.`in` } } }
```
[the remark this answers is not in the document]
Comment (ddufourboivin-zalando) [reply server/src/main/java/de/zalando/zally/rule/zalando/ProprietaryHeadersRule.kt]: Sorry! My bad. I misread then.

**Comment by ddufourboivin-zalando** [comment]
👍

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**